app/lib/attribute_cleaner.rb

```diff
@@ -0,0 +1,32 @@
+# This module provides a method for generating a configurable before_validation
+# hook for cleaning up string attributes: all listed attributes will be
+# replaced with `nil` if they are blank.
+module AttributeCleaner
```

Comment:
Looking at the implementation and the class name `AttributeCleaner` seems not quite right.  But I don't know what would be better.  Perhaps `AttributeNullifyBlanks.for`?

Reply:
Yeah, I'm also not super happy with it but also thought it's not entirely terrible. The reason I decided to leave the name fairly generic is that theoretically, we could fit more cleaning in there. I was however tempted to be extremely explicit and call it `StringAttributeCleaner`.